### PR TITLE
[3287] Send the right label in the form refresh event payload

### LINKED ISSUE #3287: The label in metadata of form representation refresh payload is not the representation name
### Steps to reproduce

1. Create a project flow
2. Create a System Form representation with a different name than the default one.
3. The 
4. Open dev-tools go to the subscription message panel and look for the message that contains a `FormRefreshEventPayload` for `formEvent`
5. Look for `data.formEvent.form.metadata.label`

### Expected result

The label value should be the representation name

### Actual result

The label value is the value returned by the `FormRepresentationDescription#titleExpression`

image

image

### PR DESCRIPTION
**Pull request by gcoutable**
# Pull request template

## General purpose
What is the main goal of this pull request?
- [x] Bug fixes
- [ ] New features
- [ ] Documentation
- [ ] Cleanup
- [ ] Tests
- [ ] Build / releng

## Project management
- [x] Has the pull request been added to the relevant project and milestone? (Only if you know that your work is part of a specific iteration such as the current one)
- [x] Have the `priority:` and `pr:` labels been added to the pull request? (In case of doubt, start with the labels `priority: low` and `pr: to review later`)
- [x] Have the relevant issues been added to the pull request?
- [x] Have the relevant labels been added to the issues? (`area:`, `difficulty:`, `type:`)
- [x] Have the relevant issues been added to the same project and milestone as the pull request?
- [ ] Has the `CHANGELOG.adoc` been updated to reference the relevant issues?
- [ ] Have the relevant API breaks been described in the `CHANGELOG.adoc`? (Including changes in the GraphQL API)
- [ ] In case of a change with a visual impact, are there any screenshots in the `CHANGELOG.adoc`? For example in `doc/screenshots/2022.5.0-my-new-feature.png`

## Architectural decision records (ADR)
- [ ] Does the title of the commit contributing the ADR start with `[doc]`?
- [ ] Are the ADRs mentioned in the relevant section of the  `CHANGELOG.adoc`?

## Dependencies
- [ ] Are the new / upgraded dependencies mentioned in the relevant section of the `CHANGELOG.adoc`?
- [ ] Are the new dependencies justified in the `CHANGELOG.adoc`?

## Frontend

This section is not relevant if your contribution does not come with changes to the frontend.

### General purpose
- [ ] Is the code properly tested? (Plain old JavaScript tests for business code and tests based on React Testing Library for the components)

### Typing
We need to improve the typing of our code, as such, we require every contribution to come with proper TypeScript typing for both changes contributing new files and those modifying existing files.
Please ensure that the following statements are true for each file created or modified (this may require you to improve code outside of your contribution).

- [ ] Variables have a proper type
- [ ] Functions’ arguments have a proper type
- [ ] Functions’ return type are specified
- Hooks are properly typed:
	- [ ] `useMutation<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useQuery<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useSubscription<DATA_TYPE, VARIABLE_TYPE>(…)`
	- [ ] `useMachine<CONTEXT_TYPE, EVENTS_TYPE>(…)`
	- [ ] `useState<STATE_TYPE>(…)`
- [ ] All components have a proper typing for their props
- [ ] No useless optional chaining with `?.` (if the GraphQL API specifies that a field cannot be `null`, do not treat it has potentially `null` for example)
- [ ] Nullable values have a proper type (for example `let diagram: Diagram | null = null;`)

## Backend

This section is not relevant if your contribution does not come with changes to the backend.

### General purpose
- [ ] Are all the event handlers tested?
- [ ] Are the event processor tested?
- [ ] Is the business code (services) tested?
- [ ] Are diagram layout changes tested?

### Architecture
- [ ] Are data structure classes properly separated from behavioral classes?
- [ ] Are all the relevant fields final?
- [ ] Is any data structure mutable? If so, please write a comment indicating why
- [ ] Are behavioral classes either stateless or side effect free?

### How to test this PR?
_Please describe here the various use cases to test this pull request_

- [ ] Has the Kiwi TCMS test suite been updated with tests for this contribution?

### CODE REVIEW

File: packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/application/controllers/RelatedElementsViewControllerIntegrationTests.java [hunk 1]
```diff
@@ -74,6 +74,7 @@ public void givenSemanticObjectWhenWeSubscribeToItsPropertiesEventsThenTheFormIs
         var flux = this.relatedElementsEventSubscriptionRunner.run(input);
 
         Predicate<Form> formPredicate = form -> {
+            System.out.println(form);
```
Comment (AxelRICHARD) [inline packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/application/controllers/RelatedElementsViewControllerIntegrationTests.java]: Could you remove this `system.out.println` please?

File: packages/forms/backend/sirius-components-collaborative-forms/src/main/java/org/eclipse/sirius/components/collaborative/forms/api/FormCreationParameters.java [hunk 2]
```diff
@@ -50,17 +52,25 @@ public String getId() {
     }
 
     public static Builder newFormCreationParameters(FormCreationParameters formCreationParameters) {
-        return new Builder(formCreationParameters.getId())
+        var builder = new Builder(formCreationParameters.getId())
                 .formDescription(formCreationParameters.getFormDescription())
                 .editingContext(formCreationParameters.getEditingContext())
                 .object(formCreationParameters.getObject())
                 .selection(formCreationParameters.getSelection());
+        if (formCreationParameters.getLabel() != null) {
```
Comment (AxelRICHARD) [inline packages/forms/backend/sirius-components-collaborative-forms/src/main/java/org/eclipse/sirius/components/collaborative/forms/api/FormCreationParameters.java]: I think this check is useless. Could you remove it please?

File: packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/application/controllers/RelatedElementsViewControllerIntegrationTests.java (hunk 1 above)
Comment (gcoutable) [reply packages/sirius-web/backend/sirius-web/src/test/java/org/eclipse/sirius/web/application/controllers/RelatedElementsViewControllerIntegrationTests.java]: Removed

File: packages/forms/backend/sirius-components-collaborative-forms/src/main/java/org/eclipse/sirius/components/collaborative/forms/api/FormCreationParameters.java (hunk 2 above)
Comment (gcoutable) [reply packages/forms/backend/sirius-components-collaborative-forms/src/main/java/org/eclipse/sirius/components/collaborative/forms/api/FormCreationParameters.java]: Check removed.

**Review by AxelRICHARD — APPROVED** [review]

**Pull request merged**